src/include/execution/compiler/operator/aggregate_util.h

```diff
@@ -0,0 +1,221 @@
+#pragma once
+
+#include <memory>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+#include "execution/compiler/operator/operator_translator.h"
+#include "planner/plannodes/aggregate_plan_node.h"
+
+namespace terrier::execution::compiler {
+
+/**
+ * Contains all information needed for an aggregation hash table..
+ */
+class AHTInfo {
+ public:
+  /**
+   * Non distinct constructor.
+   * @param codegen The code generator
+   */
+  explicit AHTInfo(CodeGen *codegen) : AHTInfo(codegen, false, 0) {}
+
+  /**
+   * Generic constructor
+   * @param codegen The code generator
+   * @param is_distinct Whether this hash table is for a distinct aggregate.
+   * @param term_idx Index of the distinct aggregate.
+   */
+  explicit AHTInfo(CodeGen *codegen, bool is_distinct, uint32_t term_idx)
+      : is_distinct_(is_distinct),
+        term_idx_(term_idx),
+        ht_(codegen->NewIdentifier("aht")),
+        struct_(codegen->NewIdentifier("AHTStruct")),
+        key_check_(codegen->NewIdentifier("ahtKeyCheckFn")),
+        entry_(codegen->NewIdentifier("aht_entry")),
+        hash_val_(codegen->NewIdentifier("aht_hash_val")) {}
+
+  /**
+   * @return Whether the hash table is for a distinct aggregate.
+   */
+  bool IsDistinct() const { return is_distinct_; }
+
+  /**
+   * @return Index of the distinct aggregate.
+   */
+  uint32_t TermIdx() const {
+    TERRIER_ASSERT(is_distinct_, "Only Valid for distinct aggregates");
+    return term_idx_;
+  }
+
+  /**
+   * @return The hash table identifier
+   */
+  const ast::Identifier &HT() const { return ht_; }
```

Comment:
None of these need to be const references, `ast::Identifiers` are const by design. Copy by value.

Reply:
By value.